### PR TITLE
Migrating WT3 crystallizer model

### PR DESCRIPTION
**Pull request by luohezhiming**
## Summary/Motivation:
Add crystallizer model

## Changes proposed in this PR:
- add crystallizer model
- add corresponding tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/unit_models/zero_order/crystallizer_zo.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    def build(self):
+        super().build()
+
+        self._tech_type = "crystallizer"
+
+        build_sido(self)
+
+        if "tds" not in self.config.property_package.solute_set:
+            raise KeyError('TDS must be included in the solute list for determining'
+                           ' electricity intensity and power consumption of the brine concentrator unit.')
```
Comment (adam-a-a) [inline watertap/unit_models/zero_order/crystallizer_zo.py]:
```suggestion
                           ' electricity intensity and power consumption of the crystallizer unit.')
```
Is this unit any different from the `brine_concentrator` unit, @luohezhiming , @kurbansitterley ?

File: watertap/unit_models/zero_order/tests/test_crystallizer_zo.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+@pytest.mark.unit
+def test_no_tds_in_solute_list_error():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.params = WaterParameterBlock(
+        default={"solute_list": ["foo"]})
+
+    with pytest.raises(KeyError,
+                       match="TDS must be included in the solute list for determining "
+                             "electricity intensity and power consumption of the brine concentrator unit."):
```
Comment (adam-a-a) [inline watertap/unit_models/zero_order/tests/test_crystallizer_zo.py]:
```suggestion
                             "electricity intensity and power consumption of the crystallizer unit."):
```

**Review by bknueven — APPROVED** [review]

**Comment by adam-a-a** [comment]
@luohezhiming this can be made into a subtype of brine_concentrator I think, but I will just create an issue for this to be revisited and approve the current changes.... unless you think you can quickly make the appropriate changes now.

**Comment by bknueven** [comment]
@luohezhiming should this be closed (replaced by #403)?

**Comment by luohezhiming** [comment]
@bknueven Yeah, we can close this PR.

**Pull request closed without merging**